examples/datasync-plugin/main.go

```diff
@@ -28,65 +28,28 @@ import (
 // ************************************************************************/
 
 // PluginName represents name of plugin.
-const PluginName = "example"
+const PluginName = "datasync-example"
 
 func main() {
-	// Start Agent with ExamplePlugin, ETCDPlugin & FlavorLocal (reused cn-infra plugins).
-	/*agent := local.NewAgent(local.WithPlugins(func(flavor *local.FlavorLocal) []*core.NamedPlugin {
-		etcdPlug := &etcd.Plugin{}
-		etcdDataSync := &kvdbsync.Plugin{}
-		resyncOrch := &resync.Plugin{}
-
-		etcdPlug.Deps.PluginInfraDeps = *flavor.InfraDeps("etcd", local.WithConf())
-		resyncOrch.Deps.PluginLogDeps = *flavor.LogDeps("etcd-resync")
-		connectors.InjectKVDBSync(etcdDataSync, etcdPlug, etcdPlug.PluginName, flavor, resyncOrch)
-
-		examplePlug := &ExamplePlugin{closeChannel: exampleFinished}
-		examplePlug.Deps.PluginInfraDeps = *flavor.InfraDeps("etcd-example")
-		examplePlug.Deps.Publisher = etcdDataSync // Inject datasync Watcher to example plugin.
-		examplePlug.Deps.Watcher = etcdDataSync   // Inject datasync Publisher to example plugin.
-
-		return []*core.NamedPlugin{
-			{etcdPlug.PluginName, etcdPlug},
-			{etcdDataSync.PluginName, etcdDataSync},
-			{resyncOrch.PluginName, resyncOrch},
-			{examplePlug.PluginName, examplePlug},
-		}
+	// Prepare ETCD data sync plugin as an plugin dependency
+	etcdDs := kvdbsync.NewPlugin(kvdbsync.UseDeps(func(deps *kvdbsync.Deps) {
+		deps.KvPlugin = &etcd.DefaultPlugin
+		deps.ResyncOrch = &resync.DefaultPlugin
 	}))
-	core.EventLoopWithInterrupt(agent, exampleFinished)*/
-
-	/*etcdPlug := etcd.NewPlugin(
-		etcd.UseConf(etcd.Config{
-			Endpoints:[]string{":1234"},
-		}),
-	)*/
-
-	//etcd.DefaultPlugin = etcd.NewPlugin(
-	//		etcd.UseConf(etcd.Config{
-	//			Endpoints:[]string{":1234"},
-	//		}),
-	//	)
-
-	etcdDataSync := kvdbsync.NewPlugin(
-		kvdbsync.UseDeps(func(deps *kvdbsync.Deps) {
-			deps.KvPlugin = &etcd.DefaultPlugin
-			deps.ResyncOrch = &resync.DefaultPlugin
-		}),
-	)
-
-	p := &ExamplePlugin{
+	// Init example plugin dependencies
+	ep := &ExamplePlugin{
 		Deps: Deps{
 			Log:          logging.ForPlugin(PluginName),
 			ServiceLabel: &servicelabel.DefaultPlugin,
-			Publisher:    etcdDataSync,
-			Watcher:      etcdDataSync,
+			Publisher:    etcdDs,
```

Comment:
We should probably keep the longer version to make this more clear to readers of example.